### PR TITLE
Style and fix account flows

### PR DESCRIPTION
**Pull request by mattleff**
Fixes https://trello.com/c/nNnMTZpG/112-style-account-reset, https://trello.com/c/wkLk6BmG/143-sync-search-active-with-design.

### CODE REVIEW

File: src/components/HOCs/withIntl.spec.tsx [hunk 1]
```diff
@@ -681,4 +681,17 @@ describe('localization usage', () => {
...
+			push: () => Promise.resolve(true),
+		});
+		loadAuthGuardData();
+		jest.spyOn(api, 'useLogout').mockResolvedValue(true);
+
+		await act(async () => {
+			const screen = await renderWithQueryProvider(<Logout />);
+			expectNoUnlocalizedText(screen);
+		});
+	});
```
Comment (narthur) [inline src/components/HOCs/withIntl.spec.tsx:696]: 👌

File: src/lib/api/useLogout.ts [hunk 2]
```diff
@@ -4,9 +4,9 @@ import { clearSessionToken } from '@lib/cookies';
 
 import { invalidateAndResetUserQueries } from '.';
 
-export function useLogout(): void {
+export function useLogout(): Promise<any> {
```
Comment (narthur) [inline src/lib/api/useLogout.ts:7]: eslint is configured to allow `any`?

Comment (mattleff) [reply src/lib/api/useLogout.ts:7]: I guess so. I'll make a card to deal with that: https://trello.com/c/RcZJacjb/144-nix-ts-anys

File: src/components/templates/andNavigation.tsx [hunk 3]
```diff
@@ -49,6 +49,8 @@ export default function AndNavigation({
 			Router.events.off('routeChangeStart', onRouteChange);
 		};
 	}, []);
+	// eslint-disable-next-line react-hooks/exhaustive-deps
+	useEffect(() => setTerm((q as string) || ''), [pathname]);
```
Comment (narthur) [inline src/components/templates/andNavigation.tsx:53]: Why are you opting out of providing exhaustive dependencies?

**Review by narthur — APPROVED** [review]
Great work!

File: src/components/templates/andNavigation.tsx (hunk 3 above)
Comment (mattleff) [reply src/components/templates/andNavigation.tsx]: Yeah, that can be simpler: 922fbda

**Pull request merged**